packages/store-ui/src/molecules/Modal/Modal.tsx

```diff
@@ -0,0 +1,112 @@
+import type {
+  AriaAttributes,
+  KeyboardEvent,
+  MouseEvent,
+  PropsWithChildren,
+} from 'react'
+import React from 'react'
+import { createPortal } from 'react-dom'
+
+import type { OverlayProps } from '../../atoms/Overlay'
+import Overlay from '../../atoms/Overlay'
+import ModalContent from './ModalContent'
+import type { ModalContentProps } from './ModalContent'
+
+interface ModalPureProps extends ModalContentProps {
+  /**
+   * ID to find this component in testing tools (e.g.: cypress, testing library, and jest).
+   */
+  testId?: string
+  /**
+   * Identifies the element (or elements) that labels the current element.
+   * @see aria-labelledby https://www.w3.org/TR/wai-aria-1.1/#aria-labelledby
+   */
+  'aria-labelledby'?: AriaAttributes['aria-label']
+  /**
+   * on Click handler for the backdrop
```

Comment:
```suggestion
   * onClick handler for the backdrop
```

Reply:
I removed it to try to apply that pattern to pass components and avoid pass props to internal components